parquet/src/test/java/org/apache/iceberg/parquet/TestParquetEncryption.java

```diff
@@ -125,4 +138,57 @@ public void testReadEncryptedFile() throws IOException {
       }
     }
   }
+
+  @Test
+  public void testReadAndWriteHadoopFile() throws IOException {
+    List<GenericRecord> records = Lists.newArrayListWithCapacity(RECORD_COUNT);
+    for (int i = 1; i <= RECORD_COUNT; i++) {
+      GenericRecord record = GenericRecord.create(SCHEMA.asStruct());
+      record.set(0, i);
+      records.add(record);
+    }
+
+    org.apache.hadoop.fs.Path path = new org.apache.hadoop.fs.Path(createTempFile(temp).toURI());
+
+    EncryptedOutputFile encryptedOutputFile =
+        EncryptionTestHelpers.createEncryptionManager()
+            .encrypt(HadoopOutputFile.fromPath(path, new Configuration()));
+    NativeEncryptionKeyMetadata keyMetadata =
+        ((NativeEncryptionOutputFile) encryptedOutputFile).keyMetadata();
+    FileAppender<GenericRecord> writer =
+        Parquet.write(encryptedOutputFile.encryptingOutputFile())
```

Comment:
I wonder why this does not result in double encryption (sending PME to AesGcmOutputStream). Seemingly, `encryptedOutputFile.encryptingOutputFile().create()` calls `new AesGcmOutputFile(..).create()`

Reply:
I don't fully understand your comment, but if you take a look at the `Parquet.write` implementation, there are different methods for Encrypted/non-Encrypted files, but all of them ends in using the `EncryptedOutputFile.encryptingOutputFile()`.

With the new implementation the `StandardEncryptedOutputFile.encryptingOutputFile()` returns the same as `StandardEncryptedOutputFile.encryptingOutputFile().encryptingOutputFile()`. Parquet uses the OutputStreams, which are also the same, no matter how many times you call the `encryptingOutputFile()` before getting the OutputStream`

Reply:
Yep, I'll need to have a deeper look, will play with the code.